test/testrunner.py

```diff
@@ -3,114 +3,52 @@
-
-
-def generate_sagemaker_pytest_cmd(image):
-    """
-    Parses the image ECR url and returns appropriate pytest command
-
-    :param image: ECR url of image
-    :return: <tuple> pytest command to be run, path where it should be executed, image tag
-    """
-    reruns = 4
-    region = os.getenv("AWS_REGION", "us-west-2")
-    integration_path = os.path.join("integration", "sagemaker")
-    account_id = os.getenv("ACCOUNT_ID", image.split(".")[0])
-    docker_base_name, tag = image.split("/")[1].split(":")
-
-    # Assign instance type
-    instance_type = assign_sagemaker_instance_type(image)
-
-    # Get path to test directory
-    find_path = docker_base_name.split("-")
-
-    # NOTE: We are relying on the fact that repos are defined as <context>-<framework>-<job_type> in our infrastructure
-    framework = find_path[1]
-    job_type = find_path[2]
-    path = os.path.join("test", "sagemaker_tests", framework, job_type)
-    aws_id_arg = "--aws-id"
-    docker_base_arg = "--docker-base-name"
-    instance_type_arg = "--instance-type"
-
-    # Conditions for modifying tensorflow SageMaker pytest commands
-    if framework == "tensorflow":
-        if job_type == "training":
-            aws_id_arg = "--account-id"
-
-            # NOTE: We rely on Framework Version being in "major.minor.patch" format
-            tf_framework_version = re.search(r"\d+(\.\d+){2}", tag).group()
-            tf_major_version = tf_framework_version.split(".")[0]
-            path = os.path.join(os.path.dirname(path), f"{framework}{tf_major_version}_training")
-        else:
-            aws_id_arg = "--registry"
-            docker_base_arg = "--repo"
-            integration_path = os.path.join(integration_path, "test_tfs.py")
-            instance_type_arg = "--instance-types"
-
-    test_report = os.path.join(os.getcwd(), "test", f"{tag}.xml")
-    return (
-        f"pytest --reruns {reruns} {integration_path} --region {region} {docker_base_arg} "
-        f"{docker_base_name} --tag {tag} {aws_id_arg} {account_id} {instance_type_arg} {instance_type} "
-        f"--junitxml {test_report}",
-        path,
-        tag,
-    )
-
-
-def run_sagemaker_pytest_cmd(image):
+def run_sagemaker_local_tests(images):
     """
-    Run pytest in a virtual env for a particular image
-
-    Expected to run via multiprocessing
-
-    :param image: ECR url
+    Function to run the SageMaker Local tests
+    :param images: <list> List of all images to be used in SageMaker tests
     """
-    pytest_command, path, tag = generate_sagemaker_pytest_cmd(image)
-
-    context = Context()
-    with context.cd(path):
-        context.run(f"virtualenv {tag}")
-        with context.prefix(f"source {tag}/bin/activate"):
-            context.run("pip install -r requirements.txt", warn=True)
-            context.run(pytest_command)
+    if not images:
+        return
+    # Run sagemaker Local tests
+    framework = images[0].split("/")[1].split(":")[0].split("-")[1]
```

Comment:
Please avoid using the ".split"-index method to get framework, job type, python version, processor type, etc. from the image uri.

Reply:
changing this to regex